src/openklant/components/klantinteracties/api/validators.py

```diff
@@ -170,12 +171,21 @@ def Rekeningnummer_exists(value):
         raise serializers.ValidationError(_("Rekeningnummer object bestaat niet."))
 
 
-class OptionalEmailValidator(EmailValidator):
-    """
-    EmailValidator for SoortDigitaalAdres that only attempts to validate if
-    `SoortDigitaalAdres` is `email`
-    """
-
-    def __call__(self, value: str, soort_digitaal_adres: str):
-        if soort_digitaal_adres == SoortDigitaalAdres.email:
-            super().__call__(value)
+class SoortDigitaalAdresValidator:
+
+    def __init__(
+        self,
+        email_validator=EmailValidator(),
+        phone_number_validator=validate_phone_number,
+    ):
+        self.email_validator = email_validator
+        self.phone_number_validator = phone_number_validator
+
+    def __call__(self, soort_digitaal_adres: SoortDigitaalAdres, value: str):
+        match soort_digitaal_adres:
+            case SoortDigitaalAdres.email:
+                self.email_validator(value)
+            case SoortDigitaalAdres.telefoonnummer:
+                self.phone_number_validator(value)
```

Comment:
Minor thing, I'll approve but I think you could remove the init and do this:

```suggestion
            case SoortDigitaalAdres.email:
                EmailValidator()(value)
            case SoortDigitaalAdres.telefoonnummer:
                phone_number_validator(value)
```

Reply:
:point_up: , I don't think the `email_validator` and the `phone_number_validator` constructor keyword arguments are going to be used as `SoortDigitaalAdres` should use the same validation consistently